Build failure: Juicefs

### Steps To Reproduce

Steps to reproduce the behavior:
1. Adding most recent package to `environment.systemPackages` from unstable
2. `sudo nixos-rebuild switch`

### Build log

```
nix log /nix/store/ank9cmcmfv3mv9v6mfyi859gqfjdw45y-juicefs-1.2.1.drv
warning: The interpretation of store paths arguments ending in `.drv` recently changed. If this command is now failing try again with '/nix/store/ank9cmcmfv3mv9v6mfyi859gqfjdw45y-juicefs-1.2.1.drv^*'
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/gjzdg4gp5abv1azzj3j95n8dnv9ganby-source
source root is source
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
Building subPackage .
# github.com/grafana/pyroscope-go/godeltaprof/internal/pprof
vendor/github.com/grafana/pyroscope-go/godeltaprof/internal/pprof/delta_mutex.go:30:20: undefined: runtime_cyclesPerSecond
vendor/github.com/grafana/pyroscope-go/godeltaprof/internal/pprof/proto.go:320:8: undefined: runtime_expandFinalInlineFrame
```

### Additional context

Using most recent version 1.2.1. #340836

### Notify maintainers

@dit7ya 

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.53, NixOS, 24.11 (Vicuna), 24.11.20241004.bc947f5`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.8`
 - channels(root): `"home-manager, nixos, nixpkgs"`
 - nixpkgs: `/home/kylepzak/.nix-defexpr/channels/nixpkgs`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

Additional info from the build failure:
```bash
error: builder for '/nix/store/ank9cmcmfv3mv9v6mfyi859gqfjdw45y-juicefs-1.2.1.drv' failed with exit code 1;
       last 10 log lines:
       > unpacking source archive /nix/store/gjzdg4gp5abv1azzj3j95n8dnv9ganby-source
       > source root is source
       > Running phase: patchPhase
       > Running phase: updateAutotoolsGnuConfigScriptsPhase
       > Running phase: configurePhase
       > Running phase: buildPhase
       > Building subPackage .
       > # github.com/grafana/pyroscope-go/godeltaprof/internal/pprof
       > vendor/github.com/grafana/pyroscope-go/godeltaprof/internal/pprof/delta_mutex.go:30:20: undefined: runtime_cyclesPerSecond
       > vendor/github.com/grafana/pyroscope-go/godeltaprof/internal/pprof/proto.go:320:8: undefined: runtime_expandFinalInlineFrame
       For full logs, run 'nix log /nix/store/ank9cmcmfv3mv9v6mfyi859gqfjdw45y-juicefs-1.2.1.drv'.
error (ignored): error: cannot unlink '/tmp/nix-build-packer-1.11.2.drv-1/build/go-cache': Directory not empty
error: 1 dependencies of derivation '/nix/store/sgfb9vl4bc1mn1k2dnvfp94hby9wmnlv-system-path.drv' failed to build
error: 1 dependencies of derivation '/nix/store/rb2j6gvbsv0f3yq708bazd1dra109vzi-nixos-system-nixos-24.11.20241014.a3c0b3b.drv' failed to build
```

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -8836,7 +8836,9 @@ with pkgs;
 
   jmespath = callPackage ../development/tools/jmespath { };
 
-  juicefs = callPackage ../tools/filesystems/juicefs { };
+  juicefs = callPackage ../tools/filesystems/juicefs {
+    buildGoModule = buildGo122Module;
+  };
 
   juicity = callPackage ../tools/networking/juicity { };
 

